x-pack/plugins/data_enhanced/common/search/poll_search.ts

```diff
@@ -42,5 +42,5 @@ export const pollSearch = <Response extends IKibanaSearchResponse>(
       takeWhile<Response>(isPartialResponse, true),
       takeUntil<Response>(aborted$)
     );
-  });
+  }).pipe(share());
```

Comment:
Share search results between subscribers!

Reply:
I think we can significantly reduce the complexity of this PR if we use `shareReplay` rather than `share`, since internally it creates its own replay subject. Then the cache can just contain what's returned from `pollSearch`, not go through the effort of creating/subscribing/managing its own subscriptions. We still have to figure out exactly when it's appropriate to `cancel`.

Is this possible or am I missing something here?